File: meetingrooms2.java
// Time Complexity : O(nlogn)
// Space Complexity : O(n)
// Did this code successfully run on Leetcode : yes
// Any problem you faced while coding this : no

// Your code here along with comments explaining your approach

class Solution {
    public int minMeetingRooms(int[][] intervals) {
        
        PriorityQueue<Integer> pq = new PriorityQueue();
        
        Arrays.sort(intervals, (a, b) -> a[0] - b[0]);
        int max = 0;
        for(int[] inter : intervals) {
                                    //30     5
            while(!pq.isEmpty() && pq.peek() <= inter[0]) {
                max = Math.max(max, pq.size());
                pq.poll();
            }
            pq.add(inter[1]);            
        }
        return max;
    }
}

File: MeetingRooms2.cs
using System;
namespace Algorithms
{
    public class MeetingRooms2
    {
        /// Time Complexity : O(nLogn) 
        // Space Complexity :O(1)
        // Did this code successfully run on Leetcode :Yes
        // Any problem you faced while coding this :  Referred Online
        public int minMeetingRooms(int[][] intervals)
        {
            PriorityQueue<Integer> pq = new PriorityQueue<Integer>();
            Arrays.sort(intervals, (a, b)->a[0] - b[0]);
            for (int[] interval : intervals)
            {
                if (!pq.isEmpty() && pq.peek() <= interval[0])
                {
                    pq.poll();
                }
                pq.add(interval[1]);
            }
            return pq.size();
        }
    }
}
